### PR TITLE
docker: Use miniforge, install from pip

### PR DESCRIPTION
**Pull request by seemethere**
Switch installation of the pytorch package to be installed from our download.pytorch.org sources which are better maintained.

As well, switching over the miniconda installation to a miniforge installation in order to ensure backwards compat for users expecting to have the conda package manager installed.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -32,7 +32,7 @@ RUN case ${TARGETPLATFORM} in \
          "linux/arm64")  MINICONDA_ARCH=aarch64  ;; \
          *)              MINICONDA_ARCH=x86_64   ;; \
     esac && \
-    curl -fsSL -v -o ~/miniconda.sh -O  "https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-${MINICONDA_ARCH}.sh"
+    curl -fsSL -v -o ~/miniconda.sh -O  "https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-${MINICONDA_ARCH}.sh"
```
Comment (seemethere) [inline Dockerfile:35]: This should probably actually be pinned to a specific version instead of latest

File: Dockerfile [hunk 2]
```diff
@@ -73,7 +73,7 @@ ARG TARGETPLATFORM
 # On arm64 we can only install wheel packages.
 RUN case ${TARGETPLATFORM} in \
          "linux/arm64")  pip install --extra-index-url https://download.pytorch.org/whl/cpu/ torch torchvision torchaudio ;; \
-         *)              /opt/conda/bin/conda install -c "${INSTALL_CHANNEL}" -c "${CUDA_CHANNEL}" -y "python=${PYTHON_VERSION}" pytorch torchvision torchaudio "pytorch-cuda=$(echo $CUDA_VERSION | cut -d'.' -f 1-2)"  ;; \
+         *)              pip install --extra-index-url https://download.pytorch.org/whl/${CUDA_VERSION#.}/ torch torchvision torchaudio ;;
```
Comment (atalman) [inline Dockerfile]: We need INSTALL_CHANNEL logic here to switch between:

https://download.pytorch.org/whl/nightly when INSTALL_CHANNEL==pytorch-nightly
https://download.pytorch.org/whl/test when INSTALL_CHANNEL==pytorch-test
https://download.pytorch.org/whl/  when INSTALL_CHANNEL==pytorch

I believe this is parameter passed from the yml workflow here: https://github.com/pytorch/pytorch/blob/main/.github/workflows/docker-release.yml#L122

**Review by malfet — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (malfet) [reply Dockerfile]: Do you just want to add another switch to rename those to `whl/nightly/`, `whl/test/` and `whl/` prefixes?

Comment (atalman) [reply Dockerfile]: Yes I think we can change the variable INSTALL_CHANNEL in https://github.com/pytorch/pytorch/blob/main/.github/workflows/docker-release.yml#L122 to reflect pypi. 
And hence here in Docker address will be: ``https://download.pytorch.org/${INSTALL_CHANNEL}/${CUDA_VERSION#.}/ ``

**Review by atalman — APPROVED** [review]
lgtm

**Pull request closed without merging**